polaris-react/src/components/RadioButton/RadioButton.scss

```diff
@@ -5,72 +5,109 @@
   margin: var(--p-space-025);
 }
 
+/* stylelint-disable selector-max-specificity, selector-max-class, selector-max-combinators -- code is much more readable this way */
 .Input {
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   @include visually-hidden;
 
   &:focus-visible + .Backdrop {
-    // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-    @include focus-ring($style: 'focused');
-
-    #{$se23} & {
-      /* stylelint-disable-next-line polaris/border/polaris/at-rule-disallowed-list -- se23 radio overrides */
-      @include no-focus-ring;
-      outline: var(--p-border-width-2) solid
-        var(--p-color-border-interactive-focus);
-      outline-offset: var(--p-space-025);
-    }
+    &,
+    // Ideally we'd use a single `:where(.ChoiceLabel:hover) &` after all these
+    // selectors to set the hover styles, but because of browser support, we
+    // have to ensure the specificity of these rules are maintained when hovered
+    // and not hovered.
+    .ChoiceLabel:hover & {
```

Comment:
We shouldn't need to add the .ChoiceLabel:hover specificity for focus-ring styles, can likely remove this.